node/state.re

```diff
@@ -235,6 +235,7 @@ let load_snapshot =
 
   // TODO: this is clearly an abstraction leak
 
+  let past_state_root_hashes = [];
```

Comment:
It seems this list will only ever grow, which is problematic. Every data structure should have some kind of upper limit.